cmake/ProjectCompilerPostConfig.cmake

```diff
@@ -0,0 +1,39 @@
+IF(${Trilinos_ENABLE_Kokkos})
+
+  # This is where to generate the gen_kokkos.cmake and KokkosCore_config.h 
+  # that we will use in the configuration
+  set(Kokkos_GEN_DIR ${CMAKE_BINARY_DIR})
+
+  # Basic initialization (Used in KOKKOS_SETTINGS)
+  set(KOKKOS_SRC_PATH ${Kokkos_SOURCE_DIR})
+  set(KOKKOS_PATH ${KOKKOS_SRC_PATH})
+
+  #------------ COMPILER AND FEATURE CHECKS ------------------------------------
+  include(${KOKKOS_SRC_PATH}/cmake/kokkos_functions.cmake)
+  set_kokkos_cxx_compiler()
+  set_kokkos_compiler_standard()
+  
+  #------------ GET OPTIONS ----------------------------------------------------
+  set(KOKKOS_CMAKE_VERBOSE True)
+  set(KOKKOS_HAS_TRILINOS True)
+  include(${KOKKOS_SRC_PATH}/cmake/kokkos_options.cmake)
+
+  #------------ COMPUTE KOKKOS_SETTINGS ----------------------------------------
+  include(${KOKKOS_SRC_PATH}/cmake/kokkos_settings.cmake)
+
+  #------------ GENERATE HEADER AND SOURCE FILES -------------------------------
+  execute_process(
+    COMMAND ${KOKKOS_SETTINGS} make -f ${KOKKOS_SRC_PATH}/core/src/Makefile build-makefile-cmake-kokkos
```

Comment:
Would this break the Windows build, or is CMake smart enough to figure out what `make` means?

Reply:
Kokkos doesn't build with MS Visual C++ yet as far as I know, but it may build on Windows in Visual Studio with the Intel compiler.

Reply:
This should work with Cygwin or the Ubuntu add-on but it would require testing.

Reply:
@kruger The projects in question use Visual Studio, not Cygwin or the Ubuntu add-on.

As far as I can tell, this would be the first bit of CMake code in Trilinos that invokes `make` directly via `execute_process`.  Does TriBITS require that `make` be in `$PATH`?

Reply:
Also, would this break the `ninja` back-end?

Reply:
@bartlettroscoe  Can you comment on make requirements?

@mhoemmen The execute_process is done at configure time and is just like invoking a script to calculate build parameters.  It's not using make for the build itself.

Reply:
I'll let @bartlettroscoe give you further direction.  I'm just pointing out potential issues.

Reply:
@kruger and @mhoemmen,

For some reason, I did not see this PR or the comments from earlier in the month.

This usage of `make` has nothing to do with the generator used by for the build of Trilinos objects.  It should work with both the Ninja and Makefile generators (and any other CMake backends).  But yet, it will require that `make` be in the path.  But to be more flexible, perhaps we should use `FIND_PROGRAM()` to to find `make` and then let the user point to a different `make`.